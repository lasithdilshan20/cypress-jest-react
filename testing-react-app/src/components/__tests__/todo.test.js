import {render,screen,cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../todo';

afterEach(cleanup);

test('Should render complete todo component',()=>{
    const todo = {id: 1, title: 'IQ Camera p1', completed: true, };
    render(<Todo todo={todo}/>);
    const todoElement = screen.getByTestId('todo-1');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('IQ Camera p1');
})

test('Should render not complete todo component',()=>{
    const todo = {id: 2, title: 'IQ Camera p2', completed: false, };
    render(<Todo todo={todo}/>);
    const todoElement = screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('IQ Camera p2');
    expect(todoElement).not.toContainHTML('<strike>');
})

test('match snapshot',()=>{
    const todo = {id: 1, title: 'IQ Camera p1', completed: false, };
    const tree = renderer.create(<Todo todo={todo}/>).toJSON();
    expect(tree).toMatchSnapshot();
})