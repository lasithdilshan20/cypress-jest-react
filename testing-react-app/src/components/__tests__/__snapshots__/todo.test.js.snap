// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    IQ Camera p1
  </h1>
</div>
`;
